core/src/main/java/org/projectnessie/cel/extension/StringsLib.java

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright (C) 2022 The Authors of CEL-Java
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.cel.extension;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+import org.projectnessie.cel.EnvOption;
+import org.projectnessie.cel.Library;
+import org.projectnessie.cel.ProgramOption;
+import org.projectnessie.cel.checker.Decls;
+import org.projectnessie.cel.interpreter.functions.Overload;
+
+/**
+ * StringsLib provides a {@link org.projectnessie.cel.EnvOption} to configure extended functions for
+ * string manipulation. As a general note, all indices are zero-based. The implementation is ported
+ * from <a href=https://github.com/google/cel-go/blob/master/ext/strings.go>cel-go</a>.
+ *
+ * <p>Note: Currently the overloading isn't supported.
+ *
+ * <p>CharAt
```

Comment:
For better readability:
* use e.g. `<h2>` or so for the function names
* surround the code examples with javadoc's `{@code ...}`
* Note that using `<` and `>` for content's not going to work - that'd be interpreted as an HTML tag and eventually not shown at all

Reply:
Done